doc/companies.md

```diff
@@ -1 +1,847 @@
 # Companies Module
+Module | HTTP Method | URL | Description
+--- | --- | --- | ---
+[Add](#add) | POST | /companies | Add Data Companies
+[View](#view) | GET | /companies/:id | View Data Companies
+[Edit](#edit) | PUT | /companies/:id | Edit Data Companies
+[Delete](#delete) | DELETE | /companies/:id | Delete Data Companies
+
```

Comment:
add endpoint for get list of companies with pagination. GET /companies 

### Param Query
Name | Example Value | Description
--- | --- | ---
search | jawara | keyword for search companies. default is empty.
page | 1 | current page. default 1.
limit | 20 | limit data in 1 pages. default 20.
order | companies.name | order of list. default companies.id
sort | asc | sort of list. option value is asc|desc. default value is desc

Example URL : GET /companies?search=jawara&order=companies.name

### Response Payload
```
{
    "status_code": "cdc-200",
    "status_message": "sucsess",
    "data": {
        "companies": [
            {
                "id":"5e8d6470-f654-24c8-93ac-332e6c3dra2e",
                "company_code": "TC",
                "name": "Test Company",
                "address": "JL. Test Company",
                "npwp": "123",
                "nomor_kukuh": "123",
                "tgl_kukuh": "2020-12-08",
                "phone_num": "021",
                "fax_num": "021",
                "seri_pajak": "12",
                "kota": "JAKARTA"
            }
        ],
        "pagination": {
            "search": "",
            "page": 1,
            "limit": 20,
            "order": "companies.name",
            "sort": "desc",
            "count": 100
        }
    }
}
```
 
there is no 404 response. If result is empty, response status code is 200, with body response payload :
```
{
    "status_code": "cdc-200",
    "status_message": "sucsess",
    "data": {
        "companies": [],
        "pagination": {
            "search": "",
            "page": 1,
            "limit": 20,
            "order": "companies.name",
            "sort": "desc",
            "count": 0
        }
    }
}
```